lib/index.js

```diff
@@ -24,6 +25,7 @@ module.exports = function (service) {
 module.exports.config = function (config, callback) {
   Hoek.assert(config && typeof config === 'object', 'Config must exist and be an object');
   Hoek.assert(config.backends && config.backends.length, 'Config must contain backends');
+  callback = callback || OrPromise();
```

Comment:
No objection to inlining it; but I was going to submit similar PR for Consulite, etc. Could end up in-lining in many modules. Which makes sense if goal is to reduce deps; but of course then some of the testing done in or-promise will need to come along for the ride as well (into potentially multiple modules). 

Could of course just wrap the guts of the function in a promise function and always return one but then you pay the "promise tax" even if you're using callbacks; which obviously isn't a big deal for these infrequently called functions. Matters for for the frequently called, and for perception... 

The advantage of or-promise is simply only having to code for the callback pattern but not requiring API consumer to have to wrap the function with silly promise shimming libs. There's other modules out there, but couldn't readily find one as simple. 

Thoughts? Happy to refactor PR either way. Would like to see promise support; not opinionated on how.